### DISCUSSION TITLE
how move cursor in insert mode

### DISCUSSION THREAD

**Discussion by hhcwyx** [ayamir/nvimdots#918] [General]
### Feature description

在输入模式下,怎么高效的引动光标?

**Comment by fecet** [comment]
I just map C-hjkl to arrows

**Comment by ayamir** [comment]
An evil idea is use `emacs` like keybind under insert mode.
FYI: https://unix.stackexchange.com/questions/49203/vim-customized-with-emacs-commands-insert-mode-only

**Comment by Jint-lzxy** [comment]
> The right way _is_ to press <kbd>Esc</kbd>, go where you want to do a small correction, fix it, go back and keep editing. It is effective because Vim has much more movements than usual character forward/backward/up/down. After you learn more of them, this will happen to be more productive.

https://stackoverflow.com/a/1737259

**Comment by fecet** [comment]
Don't really agree that use esc is the right way as it need more keystrokes(obviously) and can't generized to other occasion beyond editting like completion and command mode (they have own keymap but that require extra memory cost which is unnecessary imo) and even outside nvim.

**Comment by vollowx** [comment]
Maybe move this to discussion?

**Comment by Jint-lzxy** [comment]
I personally have <kbd>Esc</kbd> mapped to <kbd>jk</kbd>, <kbd>kj</kbd> and <kbd>jj</kbd>, so I just press these two like a single keystroke and then perform the movement I want in normal mode (this would be very convenient for non-trivial movements). For me, I dont find \<C-hjkl\> very convenient cause only the leftmost button on my keyboard is `CTRL` - my fingers need to "traverse a long distance" to reach that, but often I just want them to stay in a very limited region. If I really need to do some edit in other modes (e.g., command mode), I usually choose to use the same command as my shell instead of moving the cursor (e.g., \<C-f\> -- "ForWord", \<C-b\> -- "BackWord", \<C-w\> Del a word, etc.)

**Reply by charliie-dev** [reply]
I did pretty much the same.
And that's why we use https://github.com/rainbowhxch/accelerated-jk.nvim lol

**Comment by lkzz** [comment]
use emacs keybindings
